core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -512,10 +516,38 @@ public void kill() throws InterruptedException {
             }
             
             LOGGER.log(FINER, "Killing {0}", getPid());
+            // Firstly try to kill it gracefully, then do a forcekill if it does not help (algorithm is described in JENKINS-17116)
+            killSoftly();
             p.kill();
             killByKiller();
         }
 
+        private void killSoftly() {
+            // send Ctrl+C to the process
+            try {
+                if (!p.sendCtrlC()) {
+                    return;
+                }
+            }
+            catch (WinpException e) {
+                return;
+            }
+
+            // after that wait for it to cease to exist
+            long deadline = System.nanoTime() + softkillWaitSeconds * 1000000000;
+            do {
+                if (!p.isRunning()) {
+                    break;
+                }
+
+                try {
+                    Thread.sleep(1000);
+                }
+                catch (InterruptedException e) {
```

Comment:
InterruptedException should not be suppressed

Reply:
Done